sql/auth/sql_user.cc

```diff
@@ -3070,6 +3087,13 @@ bool mysql_drop_user(THD *thd, List<LEX_USER> &list, bool if_exists,
       return true;
     }
 
+#ifdef WITH_WSREP
+    if (start_toi_after_open_grant_tables(thd)) {
+      commit_and_close_mysql_tables(thd);
+      return true;
+    }
+#endif
```

Comment:
I think there is no need for doing this. We check if the user we want to drop is the utility user, so it will behave the same on source and replica regardless of the user executing the query.